MetaLamp/lending-pool/client/src/Capability/PollContract.purs

```diff
@@ -0,0 +1,76 @@
+module Capability.PollContract where
+
+import Prelude
+import Capability.Contract (class Contract, APIError(..), ContractId, Endpoint, callEndpoint, getContractStatus)
+import Control.Monad.Except (runExceptT, throwError)
+import Data.Either (Either(..), either)
+import Data.Generic.Rep (class Generic)
+import Data.Generic.Rep.Show (genericShow)
+import Foreign.Generic (class Decode, class Encode)
+import Halogen (HalogenM, lift)
+import Plutus.PAB.Webserver.Types (ContractInstanceClientState)
+
+class
+  Contract m <= PollContract m where
+  pollDelay :: m Unit
+  tooManyRetries :: Int -> m Boolean
+
+instance pollContractHalogenM :: PollContract m => PollContract (HalogenM st act slots msg m) where
+  pollDelay = lift pollDelay
+  tooManyRetries = lift <<< tooManyRetries
+
+data PollError
+  = TooManyRetries
+  | PollAPIError String
+  | PollResponseError String
+
+derive instance genericPollError :: Generic PollError _
+
+instance showPollError :: Show PollError where
+  show = genericShow
+
+data LeftPoll
+  = Continue
+  | ResponseError String
```

Comment:
What is the difference between `PollError` and `LeftPoll`? They have two similar constructors - `PollResponseError` and `ResponseError`, can we merge these types?